test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,543 @@
+			//check the catalogSource
+			g.By("Check OLM catalogSource, subscription, CSV and Pod")
+			waitCatalog := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				catalog, err := oc.AsAdmin().Run("get").Args("-n", "openshift-marketplace", "catalogsource", expectedSubscriptionSource, "-o=jsonpath={.status.connectionState.lastObservedState}").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if catalog != "READY" {
+					e2e.Logf("CatalogSource is not in ready state")
+					return false, nil
+				}
+				e2e.Logf("catalogSource is ready!")
+				return true, nil
+			})
+			if waitCatalog != nil {
+				e2e.Failf("catalogSource never got ready")
+			}
+
+			// check Subscription
+			waitVersion := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				csvName, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.installedCSV}").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if csvName == "" {
+					e2e.Logf("Subscription has not created the CSV, retrying")
+					return false, nil
+				}
+				e2e.Logf("The subscription is installed and the CSV is: %v", csvName)
+				return true, nil
+			})
+			if waitVersion != nil {
+				e2e.Logf("Subscription does not have an installed CSV")
+				err := oc.AsAdmin().Run("delete").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+			}
+
+			// temp workaround to extract the csvName from startingCSV field
+			csvName, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.currentCSV}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			waitCSV := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				csvStatus, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "clusterserviceversion", csvName, "-o=jsonpath={.status.phase}").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if csvStatus != "Succeeded" {
+					e2e.Logf("Cluster Service Version %s is not successful..., retrying", csvName)
+					return false, nil
+				}
+				e2e.Logf("Cluster Service Version %s has succeeded!", csvName)
+				return true, nil
+			})
+			if waitCSV != nil {
+				e2e.Failf("Cluster Service Version never got ready")
+			}
+
+			// get OLM Deployment Pod
+			waitErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				deployOSSM, err := oc.AsAdmin().CoreClient().AppsV1().Deployments(expectedSubscriptionNamespace).Get(context, "servicemesh-operator3", metav1.GetOptions{})
```

Comment:
Please check my comment in `test/extended/util/client.go`, the `CoreClient()` is not required and please use `oc.AdminKubeClient()` to replace all `oc.AsAdmin().CoreClient()`, Thanks